libgnucash/engine/gnc-numeric.cpp

```diff
@@ -207,6 +210,23 @@ numeric_from_scientific_match(smatch &m)
     return std::make_pair(num, denom);
 }
 
+static std::optional<gnc_numeric>
+fast_numeral_rational (const std::string &str)
+{
+    int64_t num, denom{};
+    auto end_ptr = str.data() + str.size();
+
+    auto [num_ptr, num_ec] = std::from_chars (str.data(), end_ptr, num);
+    if (num_ec != std::errc() || num_ptr == end_ptr || *num_ptr != '/')
```

Comment:
If `num_ptr == ent_ptr` then construct the GncNumeric with `denom = 1`.

Reply:
Ok if you want... This function was meant to speedup the default string representation. I'll insert because it's a free speedup for the "num" construct.

Reply:
> it's a free speedup

Exactly. It would be dumb to miss the opportunity.